### PR TITLE
Remove vb reference from SonarAnalyzer.CFG

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Quick fix for #9022

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs [hunk 1]
```diff
@@ -165,9 +164,9 @@ static SymbolStartAnalysisContextWrapper()
         {
             RegisterSyntaxNodeActionCS(RoslynSymbolStartAnalysisContext, action, syntaxKinds.Cast<CS.SyntaxKind>().ToImmutableArray());
         }
-        else if (languageKindType == typeof(VB.SyntaxKind))
+        else if (languageKindType.FullName == "Microsoft.CodeAnalysis.VisualBasic.SyntaxKind")
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs:167]: can we remove this? is this necessary?

Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs:167]: No. We need to keep the check. It prevents the `throw` in the else path below.

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]
can you do the updates on the RSPEC for the VB rules and include them in the PR?


- remove from SonarWay like in https://github.com/SonarSource/rspec/pull/3840/files
- deprecate like in https://github.com/SonarSource/rspec/pull/3349/files

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs [hunk 2]
```diff
@@ -135,10 +128,16 @@ static SymbolStartAnalysisContextWrapper()
             var cast = (Action<CodeBlockStartAnalysisContext<CS.SyntaxKind>>)action;
             RegisterCodeBlockStartActionCS(RoslynSymbolStartAnalysisContext, cast);
         }
-        else if (languageKindType == typeof(VB.SyntaxKind))
+        else if (languageKindType.FullName == "Microsoft.CodeAnalysis.VisualBasic.SyntaxKind")
```
[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs:131]: nevermind - it should not throw

but maybe just remove the commented out code

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs [hunk 3]
```diff
@@ -135,10 +128,16 @@ static SymbolStartAnalysisContextWrapper()
...
         {
-            var cast = (Action<CodeBlockStartAnalysisContext<VB.SyntaxKind>>)action;
-            RegisterCodeBlockStartActionVB(RoslynSymbolStartAnalysisContext, cast);
+            // Call RegisterCodeBlockStartAction for VB by:
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs:133]: I think in the end we don't want to register for VB at all

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs [hunk 4]
```diff
@@ -135,10 +128,11 @@ static SymbolStartAnalysisContextWrapper()
...
             RegisterCodeBlockStartActionCS(RoslynSymbolStartAnalysisContext, cast);
         }
-        else if (languageKindType == typeof(VB.SyntaxKind))
+        else if (languageKindType.FullName == "Microsoft.CodeAnalysis.VisualBasic.SyntaxKind")
         {
-            var cast = (Action<CodeBlockStartAnalysisContext<VB.SyntaxKind>>)action;
-            RegisterCodeBlockStartActionVB(RoslynSymbolStartAnalysisContext, cast);
+            // See https://github.com/SonarSource/sonar-dotnet/pull/9028 for how to implement this
+            // Attention: Do not add a package reference to Microsoft.CodeAnalysis.VisualBasic.Workspaces.
+            // It creates hard to detect file not found errors in .NET SDK 3 and .NET SDK 5
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs:135]: The reason is not .NET SDK 3 and 5 - they're not supported.

But they add complexity in some scenarios which we cannot fully understand...

File: analyzers/tests/SonarAnalyzer.Test/Wrappers/RegisterSymbolStartActionWrapperTest.cs [hunk 5]
```diff
@@ -18,6 +18,7 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using Microsoft.CodeAnalysis;
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/Wrappers/RegisterSymbolStartActionWrapperTest.cs:21]: this seems to be unused

Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/Wrappers/RegisterSymbolStartActionWrapperTest.cs:21]: Needed for `[ExcludeFromCodeCoverage]`

Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/Wrappers/RegisterSymbolStartActionWrapperTest.cs:21]: Hmm, that's strange. I've removed it on my local env and I can build just fine. Also, that attribute is not used in this file.

Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/Wrappers/RegisterSymbolStartActionWrapperTest.cs]: I looked in the wrong file. -> Removed.

**Review by andrei-epure-sonarsource — APPROVED** [review]
lgtm

**Comment by martin-strecker-sonarsource** [comment]
Peach validation:
* S6931: No changes
* S6934: 3 new TP in aspnet-boilerplate (the previously failing project)

**Pull request merged**